### PR TITLE
Api get timestamp to event

### PR DESCRIPTION
**Pull request by h1ppox99**

### CODE REVIEW

File: packages/matrix-client-server/src/rooms/roomId/getTimestampToEvent.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+                404,
+                errMsg(
+                  'notFound',
+                  `Unable to find event from ${params.ts} in backward direction`
+                )
+              )
+              return
+            }
+            send(res, 200, rows[0])
+          })
```
Comment (guimard) [inline packages/matrix-client-server/src/rooms/roomId/getTimestampToEvent.ts:51]: Probable missing `.catch()`

File: packages/matrix-client-server/src/rooms/roomId/getTimestampToEvent.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+                404,
+                errMsg(
+                  'notFound',
+                  `Unable to find event from ${params.ts} in forward direction`
+                )
+              )
+              return
+            }
+            send(res, 200, rows[0])
+          })
```
Comment (guimard) [inline packages/matrix-client-server/src/rooms/roomId/getTimestampToEvent.ts:77]: Probable missing `.catch()`

File: packages/matrix-identity-server/src/db/index.test.ts [hunk 3]
```diff
@@ -795,6 +795,27 @@ describe('Id Server DB', () => {
...
+                id: '2'
+              })
+              .then((rows) => {
+                expect(rows.length).toBe(0)
+                clearTimeout(idDb.cleanJob)
+                idDb.close()
+                done()
+              })
+          })
+          .catch((e) => done(e))
```
Comment (guimard) [inline packages/matrix-identity-server/src/db/index.test.ts:815]: Use `.catch(done)` _(exactly the same)_

File: packages/matrix-client-server/src/rooms/roomId/getTimestampToEvent.ts [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    // @ts-expect-error
+    const roomId: string = (req as Request).params.roomId
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-expect-error
+    const params: query_parameters = (req as Request).query
+    if (params.dir !== 'b' && params.dir !== 'f') {
+      send(res, 400, errMsg('invalidParam', 'Invalid parameters'))
+      return
+    }
+    // TODO : introduce rate-limiting
```
[the remark this answers is not in the document]
Comment (h1ppox99) [reply packages/matrix-client-server/src/rooms/roomId/getTimestampToEvent.ts:23]: I know but for the moment the rate-limit system applies to every api, event though only a few api require to be rate-limited in the spec. Is is a choice to rate-limit every api or we should specialise that in the future ?

**Pull request merged**